### ISSUE TITLE
Failed to reproduce DKD result with ResNet32x4 / ResNet8x4 on CIFAR100

### ISSUE THREAD

**Issue by cpsu00** [megvii-research/mdistiller#40]
Hello, I tried to use the default DKD settings (configs/cifar100/dkd/res32x4_res8x4.yaml) to reproduce the result of accuracy of 76.32 stated in the paper, however I was only able to get 75.8 - 76.19 as shown in the image below. 
Is there any possible reason for this? Thanks.

<img width="1078" alt="image" src="https://github.com/megvii-research/mdistiller/assets/85643374/540b13f5-6b29-4df9-b9cc-738a77fe76e2">

**Comment by Zzzzz1** [comment]
See issue #7

**Comment by yjq404** [comment]
@Zzzzz1 作者您好，我想将dkd加入到单阶段的yolov7中，请问在计算dkd_loss的要使用与target类别数量相对应的logit_student(学生预测类别置信度)和logit_teacher(老师预测类别置信度)，这时候用的logit是pred=model（imge）的logit还是最后经过NMS处理后的置信度，如果是前者的话置信度数量和target数量对不上，如果是后者的话又没有处理后的logit_teacher。希望您能指点一下，另外您觉得这样放在v7中是否有意义？祝大佬学业顺利

**Comment by Zzzzz1** [comment]
> @Zzzzz1 作者您好，我想将dkd加入到单阶段的yolov7中，请问在计算dkd_loss的要使用与target类别数量相对应的logit_student(学生预测类别置信度)和logit_teacher(老师预测类别置信度)，这时候用的logit是pred=model（imge）的logit还是最后经过NMS处理后的置信度，如果是前者的话置信度数量和target数量对不上，如果是后者的话又没有处理后的logit_teacher。希望您能指点一下，另外您觉得这样放在v7中是否有意义？祝大佬学业顺利

在detection蒸馏中dkd也是被验证有效的，可以在yolov7中使用。在DKD的论文中，我们采用的是NMS前的proposal进行蒸馏。对于YOLO类的单阶段检测器也可以这么用，使用nms前的teacher logits来蒸馏nms前的student logits。

**Comment by yjq404** [comment]
@Zzzzz1 感谢您的回复，如果是使用nms前的logits的话，区分目标类和非目标类时，yolo中预测的置信度组数远远大于target中类别数，学艺不精，您有什么好的建议吗？

**Comment by Zzzzz1** [comment]
> 

没有太理解这个置信度组数是什么意思？

**Comment by xiaohe725** [comment]
> Hello, I tried to use the default DKD settings (configs/cifar100/dkd/res32x4_res8x4.yaml) to reproduce the result of accuracy of 76.32 stated in the paper, however I was only able to get 75.8 - 76.19 as shown in the image below. Is there any possible reason for this? Thanks.
> 
> <img alt="image" width="1078" src="https://user-images.githubusercontent.com/85643374/239665879-540b13f5-6b29-4df9-b9cc-738a77fe76e2.png">

Hello, have you solved this problem? I am about 0.5 percentage points lower than the results given in the paper on all models
